src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const stateArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can use a desctructurization here for action
```suggestion
  for (const action of actions) {
    const {} = action;
    
    switch (action.type) {
```